### ISSUE TITLE
createEntityAdapter no longer excepting classes

### ISSUE THREAD

**Issue by Sawtaytoes** [reduxjs/redux#4754]
Even if this issue ends up being a non-issue, I'd like it posted online so someone else can find this solution in the future.

### Prior Issues

Not seeing any.

### What is the current behavior?

This code which previously worked:
```ts
const editorsAdapter = (
  createEntityAdapter<
    TextEditor
  >({
    selectId: (
      textEditor,
    ) => (
      textEditor
      .document
      .uri
      .toString()
    )
  })
)
```

Is now getting this error:
```ts
  Property 'id' is missing in type 'TextEditor' but required in type '{ id: EntityId; }'.ts(2344)
```

Thing is, my `selectId` function was supposed to account for that. Is that not a thing now? Did something change?

### Steps to Reproduce

Pass a generic to `createEntityAdapter` while still having a `selectId` function.

### What is the expected behavior?

Redux should be giving me a better description of how to use `createEntityAdapter for my use case. Hovering the function yields nothing:
![Image](https://github.com/user-attachments/assets/ffa117a6-e163-46cf-b0e5-bf266b12d9d1)

But Redux could document how to use it in the case of a `selectId` function and when no `selectId` function since JSDoc allows markdown, the same markdown on the website.

### Environment Details

Just recently upgraded to: `redux@^5.0.1` and noticed this issue. I was previously on `^4.2.1`.

**Comment by Sawtaytoes** [comment]
After looking through the docs, it appears the new way of doing this is by typing the `selectId` function without passing a generic:
```ts
const editorsAdapter = (
  createEntityAdapter({
    selectId: (
      textEditor: TextEditor,
    ) => (
      textEditor
      .document
      .uri
      .toString()
    )
  })
)
```

I thought passing the generic was more of an intuitive way of doing it, but you probably have your reasons for changing it.

**Comment by markerikson** [comment]
Yeah, we made changes around this in 2.0 to try to improve the typing for the ID fields.

It was easiest to infer this from `selectId`'s result if provided.
